### PR TITLE
[sw,crypto] Update the crypto API header file to align with the latest crypto library specification

### PR DESCRIPTION
**Pull request by y-srini**
The crypto API header is now updated to be in sync with the latest crypto specification [v3], as of 26Jul2022

Signed-off-by: Srinivasan Y <srinivasan.yadhunathan@seagate.com>

### CODE REVIEW

**Review by jadephilipoom — APPROVED** [review]
Looks good, thanks Srini! Just a couple of nitpicks below.

File: sw/device/lib/crypto/api.h [hunk 1]
```diff
@@ -137,10 +155,11 @@ typedef enum kdf_key_mode {
 } kdf_key_mode_t;
 
 /**
- * Enum for opentitan crypto modes using a key.
+ * Enum for opentitan crypto modes that uses a key.
```
[the remark this answers is not in the document]
Comment (y-srini) [reply sw/device/lib/crypto/api.h]: Added to the updates.

File: sw/device/lib/crypto/api.h [hunk 2]
```diff
@@ -770,28 +797,30 @@ crypto_status_t otcrypto_hash(crypto_const_uint8_buf_t input_message,
...
 /**
  * Performs the INIT operation for a cryptographic hash function.
  *
  * Initializes the generic hash context. The required hash mode is
- * selected through the `hash_mode` parameter. Populates the hash
- * context with the digest size, block size, hash update and hash
- * final APIs to be called based on the hash mode.
+ * selected through the `hash_mode` parameter. Only `kHashModeSha256`,
+ * `kHashModeSha256` and `kHashModeSha256` are supported. Other modes
+ * are not supported and an error would be returned. Populates the
```
[the remark this answers is not in the document]
Comment (y-srini) [reply sw/device/lib/crypto/api.h]: Typo, should have been kHashModeSha256, kHashModeSha384 and kHashModeSha512. Updated in the header. 

"Shouldn't all hash modes be accepted here" - The hash streaming API supports only the SHA2 modes and not the SHA3 modes. Hence the above comment.

File: sw/device/lib/crypto/api.h [hunk 3]
```diff
@@ -1039,11 +1060,34 @@ crypto_status_t otcrypto_ecdsa_keygen(drbg_state_t *drbg_state,
...
                                     crypto_const_uint8_buf_t input_message,
                                     ecc_curve_t *elliptic_curve,
                                     ecc_signature_t *signature);
 
+/**
+ * Performs the deterministic ECDSA digital signature generation.
+ *
+ * In the case of deterministic ECDSA, the random value ‘k’ for the
+ * signature generation is deterministically generated from the
+ * private key and the input message. Refer RFC6979 for details.
```
[the remark this answers is not in the document]
Comment (y-srini) [reply sw/device/lib/crypto/api.h]: Thanks, added to the update.

**Pull request merged**